packages/react/src/components/Storage/FileUploader/FileUploader.tsx

```diff
@@ -197,17 +214,19 @@ export function FileUploader({
       }
     });
 
-    const newFileStatuses = [...fileStatuses];
-    fileStatusesRef.current = newFileStatuses.map((status, index) => {
-      return {
-        ...status,
-        uploadTask: uploadTasksTemp?.[index],
-        fileState: status.fileState ?? 'loading',
-        percentage: status.percentage ?? 0,
-      };
+    setFileStatuses((prevFileStatuses) => {
+      const newFileStatuses = [...prevFileStatuses];
+      const updatedStatuses = newFileStatuses.map((status, index) => {
+        return {
+          ...status,
+          uploadTask: uploadTasksTemp?.[index],
+          fileState: status.fileState ?? 'loading',
+          percentage: status.percentage ?? 0,
+        };
+      });
+
+      return [...updatedStatuses];
```

Comment:
Syntax might be slightly off but this could just be:
```
    setFileStatuses((prevFileStatuses) => prevFileStatuses.map((status, index) => ({
      ...status,
      uploadTask: uploadTasksTemp?.[index],
      fileState: status.fileState ?? 'loading',
      percentage: status.percentage ?? 0,
    }));

Reply:
👍

Reply:
[b6c93a3](https://github.com/aws-amplify/amplify-ui/pull/3028/commits/b6c93a3be7ac04605ecb0f92846fdbaee8b465bc)